### PR TITLE
api: separate name and unique id

### PR DESCRIPTION
**Pull request by DifferentialOrange**
Separate sideservice name identifier and unique identifier. User may use both of them to search for an instance (`get(id)` or `find(name)`). User may set id explicitly or use autoincrement. Names are not required to be unique. Name is an empty string by default.

This patch also restored the number id approach from the original RFC (internal) [1].

1. https://www.notion.so/tarantool/Microservices-in-Tarantool-99657de1493946ce9f32e769e783d760

### CODE REVIEW

**Comment by DifferentialOrange** [comment]
Reminder: I plan to document public API after we establish remaining basic features.

**Review by 0x501D — APPROVED** [review]
Thank you for the patch!

File: test/integration_test.lua [hunk 1]
```diff
@@ -395,11 +399,14 @@ local function test_finding_go_http_sideservice(cg)
...
-        local found_instance = sideservice.find('go_http_server')
+        local found_instance = sideservice.get(1000)
         t.assert_equals(found_instance, known_instance)
 
+        local found_instances = sideservice.find('go_http_server')
+        t.assert_equals(found_instances, {[known_instance.id] = known_instance})
+
         local found_instances = sideservice.find()
-        t.assert_equals(found_instances['go_http_server'], known_instance)
+        t.assert_equals(found_instances, {[known_instance.id] = known_instance})
```
Comment (oleg-jukovec) [inline test/integration_test.lua:409]: I don't see a test for multiple instances with the same name, it's possible using the `find()` function.
Is it possible in practice?

**Review by oleg-jukovec — APPROVED** [review]
Thank you for the patch!

File: test/integration_test.lua (hunk 1 above)
Comment (DifferentialOrange) [reply test/integration_test.lua:409]: Sorry, I was a bit lazy with tests. Added

**Pull request merged**